iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/reports/tflint/TFLintImporter.java

```diff
@@ -0,0 +1,165 @@
+
+public class TFLintImporter extends AbstractJsonReportImporter {
+
+  private static final String MESSAGE_PREFIX = "TFLint report importing: ";
+  // Matches: `: filename.tf:2,21-29:`
+  private static final Pattern FILENAME_PATTERN = Pattern.compile(":\\s([^*&%:]+):(\\d+),(\\d+)-(\\d+):");
+
+  public TFLintImporter(SensorContext context, AnalysisWarningsWrapper analysisWarnings) {
+    super(context, analysisWarnings, MESSAGE_PREFIX);
+  }
+
+  @Override
+  protected List<JSONArray> parseJson(File reportFile) {
+    JSONObject json = null;
+    try {
+      json = (JSONObject) jsonParser.parse(Files.newBufferedReader(reportFile.toPath()));
+    } catch (IOException e) {
+      String message = String.format("could not read report file %s", reportFile.getPath());
+      logWarning(message);
+    } catch (ParseException e) {
+      String message = String.format("could not parse file as JSON %s", reportFile.getPath());
+      logWarning(message);
+    }
+    if (json == null) {
+      return Collections.emptyList();
+    }
+    JSONArray issuesArray = (JSONArray) json.get("issues");
+    JSONArray errorsArray = (JSONArray) json.get("errors");
+
+    issuesArray.addAll(errorsArray);
+
+    return List.of(issuesArray);
+  }
+
+  @Override
+  protected NewExternalIssue toExternalIssue(JSONObject issueJson) {
+    JSONObject rule = (JSONObject) issueJson.get("rule");
+    if (rule == null) {
+      // it's error, not an issue
+      String severity = (String) issueJson.get("severity");
+
+      NewExternalIssue externalIssue = context.newExternalIssue()
+        .ruleId("tflint.error")
+        .type(RuleType.CODE_SMELL)
+        .engineId("tflint")
+        .severity(severity(severity))
+        .remediationEffortMinutes(0L);
+      externalIssue.at(errorLocation(issueJson, externalIssue));
+      return externalIssue;
+    }
+
+    String ruleId = (String) rule.get("name");
+    String severity = (String) rule.get("severity");
+
+    NewExternalIssue externalIssue = context.newExternalIssue()
+      .ruleId(ruleId)
+      .type(RuleType.CODE_SMELL)
+      .engineId("tflint")
+      .severity(severity(severity))
+      .remediationEffortMinutes(0L);
+    externalIssue.at(issueLocation(issueJson, externalIssue));
+    return externalIssue;
+  }
+
+  private NewIssueLocation issueLocation(JSONObject issueJson, NewExternalIssue externalIssue) {
+    JSONObject range = (JSONObject) issueJson.get("range");
+    String filename = (String) range.get("filename");
+
+    InputFile inputFile = inputFile(filename);
+
+    JSONObject start = (JSONObject) range.get("start");
+    int startLine = asInt(start.get("line"));
+    int startColumn = asInt(start.get("column"));
+    JSONObject end = (JSONObject) range.get("end");
+    int endLine = asInt(end.get("line"));
+    int endColumn = asInt(end.get("column"));
+    TextRange textRange = inputFile.newRange(startLine, startColumn - 1, endLine, endColumn - 1);
```

Comment:
Because 2nd and 4th argument is offset, it starts from `0`. This human-readable +1 is later adjusted. The TFLint provides human-readable (1-based) so here I need to correct the value.